microsoft-edge/devtools-guide-chromium/memory-problems/dom-leaks.md

```diff
@@ -41,130 +26,326 @@ As explained in [Fix memory problems](index.md), memory issues affect page perfo
+
+1. ...
+
+
+<!-- ---------- -->
+##### [Detached Elements tool](#tab/detached-elements-tool)
+
+<!-- todo: 1: walk through these steps, clarify as needed, get familiar w old tool/steps - reduce wordcount in this numbered list -->
+
+1. Open the [Detached Elements demo webpage](https://microsoftedge.github.io/Demos/detached-elements/) in a new window or tab.
+
+   <!-- ------------------------------ -->
+   **Generate elements used by an instance of the Room class:**
+
+1. In the demo webpage, make sure that the **Room 1** button is selected.
+
+1. In the demo webpage, click the **Fast traffic** button.
+
+   The demo webpage begins generating messages and displaying them in the webpage:
+
+   ![Generating some messages in the demo webpage](./dom-leaks-images/demo-app.png)
+
+1. After some messages are displayed, click the **Stop** button in the demo webpage.
+
+   Each message is a `<div class="message">` element that's referenced by the Room 1 instance of the `Room` class.
+
+
+   <!-- ------------------------------ -->
+   **See whether there are any detached elements:**
+
+1. Right-click the webpage, and then select **Inspect**.  Or, press **Ctrl+Shift+I** (Windows, Linux) or **Command+Option+I** (macOS).  DevTools opens.
+
+1. In DevTools, in the **Activity Bar**, select the **Detached Elements** (![The Detached Elements tool icon](./dom-leaks-images/detached-elements-tool-icon.png)) tab.  If that tab isn't visible, click the **More Tools** (![More Tools icon](./dom-leaks-images/more-tools-icon.png)) button.  The **Detached Elements** tool opens:
+
+   ![Open the Detached Elements tool](./dom-leaks-images/open-detached-elements.png)
+
+1. Click the **Get Detached Elements** (![The Get Detached Elements icon](./dom-leaks-images/get-detached-elements-icon.png)) button:<!-- todo: what does clicking this button at this point accomplish or demonstrate? -->
+
+   ![0 detached elements while still in Room 1](./dom-leaks-images/0-detached-in-room-1.png)
+
+   0 detached elements are found, and no elements are displayed below the button, because all of the message elements are attached to the present, Room 1 instance of the **Room** class.
+
+
+   <!-- ------------------------------ -->
+   **Change to a different instance of the Room class, so elements become detached:**
+
+1. In the demo webpage, click the **Room 2** button.
+
+   The messages that were generated for the Room 1 instance of the **Room** class (`<div class="message">` elements) are no longer attached to the DOM, but they're still referenced by the Room 1 instance of the **Room** class.
+
+1. In DevTools, select the **Detached Elements** (![The Detached Elements tool icon](./dom-leaks-images/detached-elements-tool-icon.png)) tool, and then click the **Get Detached Elements** (![The Get Detached Elements icon](./dom-leaks-images/get-detached-elements-icon.png)) button:<!-- todo: what does clicking this button at this point accomplish or demonstrate? -->
```

Comment:
what's accomplished or demonstrated by clicking the **Get Detached Elements** button here _before_ clicking GC button, vs. clicking it _after_ GC button, a cpl steps below?
@captainbrosset 
https://review.learn.microsoft.com/en-us/microsoft-edge/devtools-guide-chromium/memory-problems/dom-leaks?branch=pr-en-us-3282&tabs=detached-elements-tool#get-detached-elements-and-analyze-a-detached-elements-javascript - step 10 vs 12; both click **Get Detached Elements** button

Reply:
This is done as a learning opportunity for readers, for them to understand that even though some elements might appear as detached at one point in time, you won't know whether they are actually memory leaks until you run GC.

When switching between chat rooms in the demo app, the web page removes the elements that are used to display messages from the DOM. But it doesn't delete these elements entirely. So they stay in memory. Then, once the GC runs (which, when using a web page like a user would, without devtools, you have no control over), some of these elements might be deleted. That's because they is no longer anything in the JavaScript code of the page that references them. So the GC can safely reclaim that memory.

What these steps demonstrate is that it's mandatory for users to force the GC to run (which, thanks to devtools, you can do) before listing the detached elements. Otherwise, you may end up with a list that shows too many elements, some that can be GC'd, and some that can't.

We could tell users to: 1. run GC 2. then get detached elements
instead of 1. get detached elements 2. run GC 3. get detached elements again

But, because the UI has 2 different buttons, instead of 1 that does both run GC and get detached elements, it would be odd to always ask users to click both buttons without telling them why.

These steps are here to tell them why they need to click both buttons.

Reply:
in article, cleaned up that explanation.  can subsequently integrate better.